simfin/utils.py

```diff
@@ -394,4 +402,50 @@ def _func_name(func):
     """
     return None if func is None else func.__name__
 
+
 ##########################################################################
+
+def into_date(date):
+    if date is None:
+        return None
+    if isinstance(date, str):
+        return datetime.strptime(date, '%Y-%m-%d')
+    if isinstance(date, datetime):
+        return date
+    else:
+        raise Exception("filters date not in format")
+
+
+def condition_function(start_date=None, end_date=None):
+    if start_date is not None:
+        if end_date is not None:
+            return lambda x: start_date < x < end_date
+        else:
+            return lambda x: start_date < x
+    else:
+        if end_date is not None:
+            return lambda x: x < end_date
+        else:
+            return None
+
+
+def filtered_file(dataset_path, start_date=None, end_date=None):
+    start_date = into_date(start_date)
+    end_date = into_date(end_date)
+
+    con_fun = condition_function(start_date, end_date)
+    # write new data file
+    new_file_name = os.path.basename(dataset_path)[0:-4] + "_filtered.csv"
+    new_file_path = os.path.join(os.path.dirname(dataset_path), new_file_name)
+
+    with open(new_file_path, 'w', newline='') as csvfile:
+        writer = csv.writer(csvfile, delimiter=';', quotechar='"')
+
+        with open(dataset_path) as csvfile:
+            datareader = csv.reader(csvfile, delimiter=';', quotechar='"')
+            for k, row in enumerate(datareader):
+                if k > 0:
```

Comment:
you have to write the first row also (without filtering) otherwise the rest doesnt work anymore